mlflow/genai/optimize/types.py

```diff
@@ -12,22 +11,61 @@
 @experimental
 @dataclass
 class PromptOptimizationResult:
+    """
+    Result of prompt optimization.
+
+    This class represents the result of :py:func:`mlflow.genai.optimize_prompt()`.
+    The optimized prompt template is automatically registered as a new version of the
+    original prompt and included in the result.
+
+    Args:
+        prompt: The prompt entity containing the optimized template.
+    """
+
     prompt: Prompt
 
 
 @experimental
 @dataclass
 class LLMParams:
-    model_name: str  # <provider>/<model name>
+    """
+    Parameters for configuring a LLM Model.
+
+    Args:
+        model_name: Name of the model in the format `<provider>/<model name>`.
+            For example, "openai/gpt-4" or "anthropic/claude-4".
+        base_uri: Optional base URI for the API endpoint. If not provided,
+            the default endpoint for the provider will be used.
+        temperature: Optional sampling temperature for the model's outputs.
+            Higher values (e.g., 0.8) make the output more random,
+            while lower values (e.g., 0.2) make it more deterministic.
+    """
+
+    model_name: str
     base_uri: Optional[str] = None
     temperature: Optional[float] = None
 
 
 @experimental
 @dataclass
 class OptimizerConfig:
+    """
+    Configuration for prompt optimization.
+
+    Args:
+        num_instruction_candidates: Number of candidate instructions to generate
+            during each optimization iteration. Higher values may lead to better
+            results but increase optimization time. Default: 8
+        max_few_show_examples: Maximum number of examples to show in few-shot
+            demonstrations. Default: 3
+        num_threads: Number of threads to use for parallel optimization.
+        optimizer_llm: Optional LLM parameters for the teacher model. If not provided,
+            the target LLM will be used as the teacher.
+        algorithm: The optimization algorithm to use. Default: "DSPy/MIPROv2"
+    """
+
     num_instruction_candidates: int = 8
     max_few_show_examples: int = 3
-    num_threads: int = ((multiprocessing.cpu_count() or 1) * 2) + 1
+    num_threads: Optional[int] = None
```

Comment:
Moved the default thread counts into optimizer logic as `((multiprocessing.cpu_count() or 1) * 2) + 1` is evaluated and the actual number appeared in the api doc.

Reply:
That worked too, thanks